### PR TITLE
ci: setup initial release process

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds pr linting GH action
- adds release please GH action

### Notes

This won't actually publish the Gem but includes all the plumbing.

### Follow-up Tasks

- Extend the release step to publish the Gem
- https://github.com/open-feature/ruby-sdk-contrib/issues/7

### How to test

We'll do it live!

### CODE REVIEW

File: .github/workflows/release-please.yaml
Comment (beeme1mr) [inline .github/workflows/release-please.yaml:14]: I left this on v3 because v4 still has a few issues.

File: .github/workflows/release-please.yaml
Comment (beeme1mr) [inline .github/workflows/release-please.yaml:23]: Debugging statement

File: providers/openfeature-flagd-provider/lib/openfeature/flagd/provider/version.rb
Comment (beeme1mr) [inline providers/openfeature-flagd-provider/lib/openfeature/flagd/provider/version.rb:7]: No idea if this is necessary or valid. It was the pattern we used in the Ruby SDK and it looks like Release Please looks for a `version.rb` file.

File: .release-please-manifest.json
Comment (beeme1mr) [inline .release-please-manifest.json:3]: This maps to the folder structure. We should try and align on a naming convention before pushing the Gems.

File: providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec
Comment (beeme1mr) [inline providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec:3]:
```suggestion

require_relative "lib/open_feature/sdk/provider/version"
```

Comment (beeme1mr) [reply providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec:3]: This repo uses `openfeature` and `open_feature` as a folder name in various places. I don't have a preference for what we use but we should stay consistent. @maxveldink any preference?

File: providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec
Comment (beeme1mr) [inline providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec:19]: The changelog will be automagically created and managed by release please.

File: release-please-config.json
Comment (beeme1mr) [inline release-please-config.json:14]:
```suggestion
      "version-file": "lib/open_feature/sdk/provider/version.rb",
```

File: release-please-config.json
Comment (beeme1mr) [inline release-please-config.json:13]:
```suggestion
      "package-name": "openfeature-sdk-meta_provider",
```

Comment (beeme1mr) [reply release-please-config.json:13]: @maxveldink how would you feel about `openfeature-meta-provider` or something similar? I think we should drop the `sdk` portion.

File: .github/workflows/release-please.yaml
Comment (alxckn) [inline .github/workflows/release-please.yaml:1]: nit: use `.yml` extension to be consistent

**Review by alxckn — COMMENTED** [review]
Not familiar with release-please, what is the intended workflow? For release please to maintain a PR as described on its repo that once merged will proceed with the release?

Will it require that each provider gets released at the same time?

File: release-please-config.json
Comment (beeme1mr) [inline release-please-config.json:2]: @alxckn this setting configures release please to open separate prs for each artifact.

File: providers/openfeature-sdk-meta_provider/lib/open_feature/sdk/provider/version.rb
Comment (maxveldink) [inline providers/openfeature-sdk-meta_provider/lib/open_feature/sdk/provider/version.rb:1]: We were actually just wrestling with this in an internal provider gem about where to put this file. It's weird to me that it would be `OpenFeature::SDK::Provider::VERSION`, because you potentially are going to have a runtime collision with other providers that are versioned the same way. The constant should probably be something like `OpenFeature::SDK::Provider::META_PROVIDER_VERSION`. Does `release-please` expect it to only be named `VERSION`?

Comment (maxveldink) [reply providers/openfeature-sdk-meta_provider/lib/open_feature/sdk/provider/version.rb:1]: The other idea would be to have this version file _not_ be required by the gem itself, but only be used in the `gemspec` to control the version.

**Review by maxveldink — COMMENTED** [review]
@beeme1mr I'm glad this is a forcing function to address the naming inconsistencies we have. Here's some thoughts:
* Conventionally in Ruby, you'll see snake_case files map to PascalCase constants. So, if we want the `OpenFeature` to be our top-level module, you'd expect every path to be `open_feature`. We've been talking about this in [this issue](https://github.com/open-feature/ruby-sdk/issues/71) on the SDK. My preference is we do that here.
* This runs into an issue with the SDK artifact we're currently releasing. `openfeature-sdk` would result in an expected constant of `Openfeature::Sdk`. Most folks will conventionally know that `Sdk` should be inflected to `SDK`, so that's fine, but the top-level module is still named unexpectedly. I have thought that it might make sense to rename the primary package to `open_feature` before we release `1.0` (also dropping SDK because I don't think it adds value for us), but I didn't know how much will there would be for that.
* Bringing that all into these gems, I think my desired naming would be `open_feature-meta_provider` and `open_feature-flag_d` (of `open_feature-flag_d_provider` if we're _only_ shipping a provider). That matches gem naming conventions where the `-` indicates that something is modifying/plugging into another gem and has the expected casing.
* I am but one voice in this, and happy to have the discussion. There's plenty of non-conventional gem names/module pathing and it's fine as long as you explain usages in your README.

File: .github/workflows/release-please.yaml
Comment (maxveldink) [reply .github/workflows/release-please.yaml:14]: Can you point me toward what the issues with v4 are? Do we know if the maintainers have it prioritized/are working on it?

File: providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec
Comment (maxveldink) [reply providers/openfeature-sdk-meta_provider/openfeature-sdk-meta_provider.gemspec]: After reviewing what our other language repos were doing, I pushed up two commits this morning with some preference. I think we should go with `openfeature`, and I simplified the MetaProvider structure. While we're pre-1, I'll open up a PR to adjust the primary repo again to match this decision.

**Comment by maxveldink** [comment]
@beeme1mr I made some modifications to the naming for the meta_provider (and I'll have some follow-on changes in the ruby-sdk). I like the place we ended up 👍🏻

File: .github/workflows/release-please.yaml
Comment (beeme1mr) [reply .github/workflows/release-please.yaml]: There are a few minor issues that we could likely address. The first relates to the releases_created [output](https://github.com/google-github-actions/release-please-action/issues/912). You also can't have the bot sign off on a comment. This fix will come once [this release](https://github.com/google-github-actions/release-please-action/pull/969) is merged.

I'm not sure if there are other issues. I moved away from v4 once I discovered the limited listed above.

**Comment by beeme1mr** [comment]
> @beeme1mr I made some modifications to the naming for the meta_provider (and I'll have some follow-on changes in the ruby-sdk). I like the place we ended up 👍🏻

Awesome, looks good to me! Are we good to merge? If this works, it should create release PRs, and merging those PRs should do everything except publish the Gem. If that goes smoothly, I think it should be pretty straightforward to finish up the publishing step.

**Comment by maxveldink** [comment]
Yep! I'll try and merge this in first thing tomorrow.

**Review by maxveldink — APPROVED** [review]

**Pull request merged**
